controllers/secret_controller.go

```diff
@@ -1003,6 +1069,44 @@ func (r *SecretReconciler) getClusterProxy() (string, error) {
 	return "", nil
 }
 
+// isManagementCluster checks if the current cluster is a HyperShift management cluster
+// by examining the infrastructureName in the Infrastructure object
+func (r *SecretReconciler) isManagementCluster() (bool, error) {
+	var infra configv1.Infrastructure
+	err := r.Client.Get(context.TODO(), client.ObjectKey{Name: "cluster"}, &infra)
+	if err != nil {
+		return false, err
+	}
+
+	// Primary detection: Check if infrastructureName starts with "hs-mc"
+	// HyperShift management clusters have infrastructure names like "hs-mc-<cluster-id>-<hash>"
+	// Example: "hs-mc-ibv8l52c0-d4v8v"
+	if infra.Status.InfrastructureName != "" {
+		if len(infra.Status.InfrastructureName) >= 5 {
+			// Check if the name starts with "hs-mc" (first 5 characters)
+			if infra.Status.InfrastructureName[:5] == "hs-mc" {
+				return true, nil
+			}
+		}
+	}
+
+	// Fallback: Check via ClusterVersion annotations
+	var version configv1.ClusterVersion
+	err = r.Client.Get(context.TODO(), client.ObjectKey{Name: "version"}, &version)
+	if err != nil {
+		// If we can't get the ClusterVersion, don't fail - just return false
+		return false, nil
+	}
+
+	// Check if cluster version has hypershift annotations
+	if clusterType, exists := version.Annotations[clusterTypeLabelKey]; exists {
+		return clusterType == clusterTypeManagement, nil
+	}
+
+	// Default to false if not a management cluster
+	return false, nil
```

Comment:
This is the same decision case as the ability to read the CM, above.  

What do we want to have happen if we accidentally cannot tell?  Chance that we assume it's not an MC, and we miss alerts, or that we accidentally page for DT namespaces on a non-MC cluster?

Which is more desirable?

Reply:
The implementation defaults to `readMC = true` (management cluster behavior) when cluster type cannot be determined. This is a fail-safe decision that prioritizes safety:

  **Rationale:**
  - **If we default to false (not MC) and we're wrong:** We miss critical MC namespace alerts (e.g., `hosted-*`, HyperShift control plane namespaces). This could lead to undetected incidents on actual management clusters.
  - **If we default to true (MC) and we're wrong:** We monitor some additional namespaces on a non-MC cluster. This creates noise but doesn't miss any critical alerts.

  **Risk Assessment:**
  - Missing MC alerts = High severity (undetected incidents)
  - Extra monitoring on non-MC = Low severity (noise, easily identifiable in logs/metrics)